### PR TITLE
Sped up higher layer build 

### PR DESCRIPTION
**Pull request by sdorkenw**

### CODE REVIEW

File: pychunkedgraph/backend/chunkedgraph.py [hunk 1]
```diff
@@ -491,14 +498,29 @@ def get_unique_node_id(self, chunk_id: np.uint64) -> np.uint64:
...
-                                        counter_key, 1)
+                                        counter_key, step)
 
         # This increments the row entry and returns the value AFTER incrementing
         latest_row = append_row.commit()
         segment_id_b = latest_row[self.incrementer_family_id][counter_key][0][0]
         segment_id = int.from_bytes(segment_id_b, byteorder="big")
 
-        return self.get_node_id(np.uint64(segment_id), chunk_id=chunk_id)
+        return np.uint64(segment_id)
```
[the remark this answers is not in the document]
Comment (sdorkenw) [reply pychunkedgraph/backend/chunkedgraph.py]: For backward compatibility reasons it does not. It only returns the highest value of the range.

File: pychunkedgraph/backend/chunkedgraph.py [hunk 2]
```diff
@@ -351,8 +351,12 @@ def get_serialized_info(self):
         """
         info = {"table_id": self.table_id,
                 "instance_id": self.instance_id,
-                "project_id": self.project_id,
-                "credentials": self.client.credentials}
+                "project_id": self.project_id}
+
+        try:
+            info["credentials"] = self.client.credentials
+        except:
```
[the remark this answers is not in the document]
Comment (sdorkenw) [reply pychunkedgraph/backend/chunkedgraph.py]: That is a fix related to what was tried in this PR

File: pychunkedgraph/backend/chunkedgraph.py [hunk 3]
```diff
@@ -491,14 +498,29 @@ def get_unique_node_id(self, chunk_id: np.uint64) -> np.uint64:
...
+        atomic counter
+
+        :param chunk_id: np.uint64
+        :param step: int
+        :return: np.uint64
+        """
+
+        segment_id = self.get_unique_segment_id(chunk_id=chunk_id, step=step)
+
+        return self.get_node_id(segment_id, chunk_id=chunk_id)
```
[the remark this answers is not in the document]
Comment (sdorkenw) [reply pychunkedgraph/backend/chunkedgraph.py]: see above, only one id returned

File: pychunkedgraph/backend/chunkedgraph.py [hunk 4]
```diff
@@ -830,12 +846,53 @@ def range_read_chunk(self, layer: int, x: int, y: int, z: int,
...
+            rows *with* this column will be ignored
+        :param time_stamp: datetime.datetime
+        :param yield_rows: bool
+        :return: list or yield of rows
+        """
+        chunk_id = self.get_chunk_id(layer=layer, x=x, y=y, z=z)
+        max_segment_id = self.get_segment_id_limit(chunk_id)
+
+        # Define BigTable keys
+        start_id = self.get_node_id(np.uint64(0), chunk_id=chunk_id)
```
[the remark this answers is not in the document]
Comment (sdorkenw) [reply pychunkedgraph/backend/chunkedgraph.py]: True, I guess that is still left from a time when we did not have that convention

File: pychunkedgraph/backend/chunkedgraph.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdorkenw) [reply pychunkedgraph/backend/chunkedgraph.py]: That's actually also how BigTable does it. I see that it is more convenient to return the range but there might be too much code dependent on this function returning a single id (?)

File: pychunkedgraph/backend/chunkedgraph.py [hunk 5]
```diff
@@ -1157,11 +1214,14 @@ def add_atomic_edges_in_chunks(self, edge_id_dict: dict,
...
         ccs = list(nx.connected_components(chunk_g))
 
-        # print("CC in chunk: %.3fs" % (time.time() - time_start))
+        print("CC in chunk: %.3fs" % (time.time() - time_start))
 
         # Add rows for nodes that are in this chunk
         # a connected component at a time
         node_c = 0  # Just a counter for the print / speed measurement
+
+        u_id_range_start = self.get_unique_segment_id(parent_chunk_id, len(ccs)) - np.uint64(len(ccs) - 1)
```
[the remark this answers is not in the document]
Comment (sdorkenw) [reply pychunkedgraph/backend/chunkedgraph.py]: I agree -- same issue as above

File: pychunkedgraph/backend/chunkedgraph.py [hunk 6]
```diff
@@ -1276,91 +1316,91 @@ def add_atomic_edges_in_chunks(self, edge_id_dict: dict,
...
+
+                    row_ids = row_ids[column_ids == 0]
+                    column_ids = column_ids[column_ids == 0]
+                    inv_column_ids = (column_ids + 1) % 2
+                    time_dict["cross_mask"].append(time.time() - time_start_2)
+                    time_start_2 = time.time()
+
+                    connected_ids = np.concatenate([connected_ids, edge_id_dict["cross"][row_ids, inv_column_ids]])
+                    connected_affs = np.concatenate([connected_affs, np.full((len(row_ids)), np.inf, dtype=np.float32)])
+                    connected_areas = np.concatenate([connected_areas, np.full((len(row_ids)), np.inf, dtype=np.uint64)])
```
Comment (nkemnitz) [inline pychunkedgraph/backend/chunkedgraph.py]: `np.inf` can not be represented by a `np.uint64`. Numpy is silently doing an unsafe cast [here](https://github.com/numpy/numpy/blob/v1.15.1/numpy/core/numeric.py#L310) and you miraculously end up with `0`. But that's not something you can/should rely on.

File: pychunkedgraph/backend/chunkedgraph.py [hunk 7]
```diff
@@ -34,6 +36,31 @@
            HOME + "/.cloudvolume/secrets/google-secret.json"
 
 
+def compute_sparse_indices(arr: np.ndarray) -> sparse.csr_matrix:
```
Comment (nkemnitz) [inline pychunkedgraph/backend/chunkedgraph.py]: Would be nice if we can throw out at least one of those big dependencies.

**Comment by sdorkenw** [comment]
The latest version is ready to be merged into the master. All tests pass (one warning).

**Pull request merged**